addons/mail/models/res_users.py

```diff
@@ -269,42 +269,69 @@ def _init_messaging(self):
     @api.model
     def systray_get_activities(self):
         activities = self.env["mail.activity"].search([("user_id", "=", self.env.uid)])
-        activities_by_record_by_model_name = defaultdict(lambda: defaultdict(lambda: self.env["mail.activity"]))
-        for activity in activities:
-            record = self.env[activity.res_model].browse(activity.res_id)
-            activities_by_record_by_model_name[activity.res_model][record] += activity
-        model_ids = list({self.env["ir.model"]._get(name).id for name in activities_by_record_by_model_name.keys()})
-        user_activities = {}
-        for model_name, activities_by_record in activities_by_record_by_model_name.items():
-            domain = [("id", "in", list({r.id for r in activities_by_record.keys()}))]
-            allowed_records = self.env[model_name].search(domain)
-            if not allowed_records:
+
+        # 1. Sorting the activities in model > module > record
```

Comment:
Note this method is mostly rewritten in bugfix https://github.com/odoo/odoo/pull/134126 that is not yet landed in master, this should probably be done in a way compatible with it.

Reply:
> Note this method is mostly rewritten in bugfix #134126 that is not yet landed in master, this should probably be done in a way compatible with it.

Thank you for your review and your feedback! I've had a look at this PR and it changes quite a bit the logic of this method. I understand why it is needed but at the moment I neither have the time, nor the energy to continue to adapt this code to other PR while they are created.

I've created another PR that imitates the behavior of mass_mailing to split activities for Project/To-Do that should be sufficient functionally for us. I still think that the code of this PR could be a useful refactoring as it avoids to add extra requests on top of what is done in mail. So if you feel that it could be useful in the future, don't hesitate to use my code.